### PR TITLE
MINOR: Use debug level logging for noisy log messages in Connect

### PR DESCRIPTION
**Pull request by cyrusv**
This very simply reduces the log level for these very frequently logged, very infrequently useful log messages to debug level.
My experience operating these connectors is that these logs make reading info level logs quite cluttered. The MDC context certainly helps filter, but I think debug is more appropriate here -- if the log level were debug and someone wanted to promote it to info, that certainly would be an odd desire!

Debug is recommended here over Trace, since these do not really help trace specific messages or paths, and give kind of general debug information at a regular cadence (volume of these log messages does not grow with throughput). That said, if there is strong support for Trace I think that would be an improvement over Info, too.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by cyrusv** [comment]
@wicknicks, @C0urante, @ryannedolan:
Any feedback on this one? It will make logs on many systems much less expensive in the normal case.

**Comment by ryannedolan** [comment]
makes sense to me, thanks

**Review by C0urante — APPROVED** [review]
This is definitely a step forward. I have a bikeshedding comment but overall I'm in favor so ✅ from me.

Bikeshedding comment: generating multiple `DEBUG`-level messages for a single offset commit (which is what these changes would do for the `WorkerSourceTask` class) seems unnecessary. I think at most one of those should be `DEBUG` and the remainder should be `TRACE`.

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java
Comment (wicknicks) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java:346]: these shouldn't be noisy. there is some [dedup logic](https://github.com/apache/kafka/blob/2.5.0/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java#L430-L434) that should prevent the same message from being printed over and over again.

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java
Comment (wicknicks) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java:478]: nit: since this is indicating entry into a method, I think it can be trace level.

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java
Comment (wicknicks) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java:574]: nit: since this is mostly indicating exit from the method, I think it can be trace level too.

**Review by wicknicks — COMMENTED** [review]
thanks for the improvement, @cyrusv. agree that we need to keep the logs quieter for the source connectors' offset commits.

**Comment by cyrusv** [comment]
@C0urante, @wicknicks -- I downgraded two of the SourceTask even further to Trace, so only the central log statement is debug level. 

@wicknicks I see your point about dedup in SinkTask, but even with dedup it's still just not that useful of a log in my opinion to warrant Info level. Happy to discuss more if you disagree and think

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java
Comment (wicknicks) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java:574]: I think you missed this one.

**Review by wicknicks — APPROVED** [review]
thanks, @cyrusv! LGTM, pending one comment.

**Comment by cyrusv** [comment]
@kkonstantine or @rhauch  -- This has some traction and nonbinding approval -- I think we're ready for committer review?

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java
Comment (rhauch) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java]: Why set this to `trace` rather than `debug` like the other changes? Isn't it useful to know that we've entered this method before we lock?

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java
Comment (rhauch) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java]: Same question here about `trace` vs `debug`, as denoted above. With debug logs, isn't it useful to know how long it took to commit all offsets?

**Review by rhauch — COMMENTED** [review]
Are we sure that it's never useful at INFO level to know when offsets are committed? Normally, the offset commit interval is not terribly frequent (`offset.flush.interval.ms` defaults to 1 minute).

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java
Comment (cyrusv) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java]: @rhauch, The duration is an interesting point. I was considering that this is tracing exit of a function which already has debug in it. I think the duration is an aspect I hadn't considered, so I'll be happy to make these all uniformly debug level

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java
Comment (rhauch) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java]: I'm not convinced that changing this to DEBUG is worth it. The number of source records output in this log message can be instrumental in some cases.

For example, if the producer does not keep up with the source task (for whatever reason), this currently INFO-level message appears shortly before the following ERROR-level message:
```
... ERROR WorkerSourceTask{id=...} Failed to commit offsets (org.apache.kafka.connect.runtime.SourceTaskOffsetCommitter)
```
The number of outstanding source records reported by this line is an important factor in determining how to tune the producer and `offset.flush.timeout.ms` value.

Comment (cyrusv) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java]: I do not feel strongly about this particular message -- it represents about 2% of the connect log volume in my deployment, so not too bad. Are you still onboard with the other log levels I've updated in the PR? In which case, I will revert this one?

Comment (rhauch) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java]: Yeah, I think the log messages other than this one are fine as DEBUG.

Comment (cyrusv) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java]: Sounds good, updated

**Comment by cyrusv** [comment]
@rhauch, thanks for the review, I've made the requested changes.

**Review by rhauch — APPROVED** [review]
LGTM. Thanks, @cyrusv

**Pull request merged**
